### PR TITLE
hamid/setup-eslint

### PR DESCRIPTION
**Pull request by hamid-deriv**
# Description

Setup Eslint and its related packages in Deriv-Components

Addresses (redmine-issue-number)#

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update
-   [ ] Package update

# Author Checklist:

Please check relevant items before requesting for review.

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ module.exports = {
         'plugin:@typescript-eslint/recommended',
         'prettier/@typescript-eslint',
         'plugin:prettier/recommended',
+        // For disabling react/react-in-jsx-scope
+        'plugin:react/jsx-runtime',
```
Comment (balakrishna-binary) [inline .eslintrc.js:9]: `deriv-app` still uses react 16 and we don't bundle react along with `@deriv/ui` package. So, we should not include this plugin yet - `plugin:react/jsx-runtime`.

Comment (balakrishna-binary) [reply .eslintrc.js:9]: We expect `react` to be a peer dependency. When we use this package in `deriv-app`, it would pick react16 instead.

**Comment by balakrishna-binary** [comment]
we should not be removing package-lock.json

File: .eslintrc.js (hunk 1 above)
Comment (balakrishna-binary) [reply .eslintrc.js:9]: I think it's fixed here - https://github.com/binary-com/deriv-components/pull/261/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R86

**Comment by hamid-deriv** [comment]
> we should not be removing package-lock.json

This file is not removed. It just changed.

File: .eslintrc.js (hunk 1 above)
Comment (hamid-deriv) [reply .eslintrc.js:9]: Sorted.

**Comment by balakrishna-binary** [comment]
@hamid-deriv ah yeah. It looks like there are a lot of changes in package-lock.json. I think it's changed to lockfileVersion 1

**Review by balakrishna-binary — APPROVED** [review]

**Review by yashim-deriv — APPROVED** [review]

**Pull request merged**